File: backend/app/schemas/sentiment.py
from pydantic import BaseModel, Field
from typing import Literal


class AnalyzeRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000, description="Texto a analizar")

class EmotionScores(BaseModel):
    joy: float = Field(ge=0, le=100)
    anger: float = Field(ge=0, le=100)
    sadness: float = Field(ge=0, le=100)
    fear: float = Field(ge=0, le=100)
    surprise: float = Field(ge=0, le=100)
    disgust: float = Field(ge=0, le=100)

class SentimentResult(BaseModel):
    sentiment: Literal["positive", "negative", "neutral", "mixed"]
    score: float = Field(ge=-1.0, le=1.0)
    confidence: float = Field(ge=0, le=1)
    emotions: EmotionScores
    keywords: list[str] = Field(min_length=1, max_length=5)
    summary: str

File: frontend/src/types/sentiment.ts
export type AnalyzeRequest = {
    text: string
}

export type EmotionScores = {
    joy: number;
    anger: number;
    sadness: number;
    fear: number;
    surprise: number;
    disgust: number
}

export type SentimentType ="positive" | "negative" | "neutral" | "mixed"

export interface SentimentResult {
    sentiment: SentimentType;
    score: number;
    confidence: number;
    emotions: EmotionScores;
    keywords: string[];
    summary: string
}

export interface HistoryEntry {
    text: string;
    result: SentimentResult;
    ts: number
}

export interface TextInputProps {
    text: string;
    loading: boolean;
    onChange: (value: string) => void;
    onAnalyze: () => void
}

export interface ResultCardProps {
    result: SentimentResult,
    config: {
        color: string;
        label: string;
        bg: string
    }
}

export interface KeywordPillsProps {
    keywords: string[];
    sentiment: SentimentType
}

export interface HistoryCardProps {
    history: HistoryEntry[];
    selected: HistoryEntry | null;
    onSelect: (entry: HistoryEntry) => void;
    onClear: () => void
}

export interface HistoryItemProps {
    item: HistoryEntry;
    index: number;
    isSelected: boolean;
    onSelect: (entry: HistoryEntry) => void;
}

export interface StatsBarProps {
    total: number;
    positive: number;
    negative: number;
    avgScore: number | null
}